xbmc/Application.cpp

```diff
@@ -4161,6 +4165,50 @@ void CApplication::ShowAppMigrationMessage()
   }
 }
 
+void CApplication::ConfigureAndEnableAddons()
+{
+  std::vector<std::shared_ptr<IAddon>>
+      disabledAddons; /*!< Installed addons, but not auto-enabled via manifest */
+
+  if (CServiceBroker::GetAddonMgr().GetDisabledAddons(disabledAddons) && !disabledAddons.empty())
+  {
+    // only look at disabled addons with disabledReason == NONE
+    // usually those are installed from package managers or manually.
+    // also omit addons of type dependency
+
+    for (const auto& addon : disabledAddons)
+    {
+      if (CServiceBroker::GetAddonMgr().IsAddonDisabledExcept(addon->ID(),
+                                                              ADDON::AddonDisabledReason::NONE) ||
+          CAddonType::IsDependencyType(addon->MainType()))
+        continue;
+
+      if (HELPERS::ShowYesNoDialogText(CVariant{24059}, addon->Name()) ==
+          DialogResponse::YES) // Would you like to enable this add-on?
+      {
+        if (addon->HasSettings())
+        {
+          if (CGUIDialogAddonSettings::ShowForAddon(addon))
+          {
+            // only enable if settings dialog hasn't been cancelled
+            CServiceBroker::GetAddonMgr().EnableAddon(addon->ID());
+          }
+        }
+        else
+        {
+          CServiceBroker::GetAddonMgr().EnableAddon(addon->ID());
+        }
+      }
+      else
+      {
+        // user chose not to configure/enable so we're not asking anymore
+        CServiceBroker::GetAddonMgr().UpdateDisabledReason(addon->ID(),
+                                                           ADDON::AddonDisabledReason::USER);
```

Comment:
So many CServiceBroker::GetAddonMgr() in this method, used even in a loop... Would it make sense to define a local varibale holding the addon manager?